### PR TITLE
add type hint for 'data' argument in DataFrame's from_dict method: Un…

### PR DESCRIPTION
**Pull request by GrossBetruger**
…ion[dict, list]

- [ ] closes #xxxx
- [ ] tests added / passed
- [ ] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/core/frame.py [hunk 1]
```diff
@@ -1458,7 +1458,7 @@ def __rmatmul__(self, other):
     @classmethod
     def from_dict(
         cls,
-        data,
+        data: Union[dist, list],
```
Comment (attack68) [inline pandas/core/frame.py:1461]:
```suggestion
        data: dict | list,
```

Comment (GrossBetruger) [reply pandas/core/frame.py]: done

**Review by simonjayhawkins — CHANGES_REQUESTED** [review]
Thanks @GrossBetruger for the PR. this is failing type validation and is inconsistent with the docstring. can you fixup to get mypy passing.

**Comment by simonjayhawkins** [comment]
@GrossBetruger Thanks for the PR. closing as stale. ping if you want to continue.

**Pull request closed without merging**